docs/contract/dataplanes-dataframes.md

```diff
@@ -0,0 +1,99 @@
+---
+id: contract
+title: Grafana data structure 
+description: Dataplanes and Dataframes.
+keywords:
+  - data-plane-contract
+  - dataplane
+  - dataframe
+slug: /  
+---
+
+# Grafana data structure
+
+Grafana supports a variety of different data sources, each with its own data model. To make this possible, Grafana consolidates the query results from each of these data sources into one unified data structure called a **data frame**. The **data plane** is built on top of the data frame and includes additional information on what the data frame holds. 
+
+## Data frames overview  
+
+A data frame is a data structure that consolidates the query results from your data sources, providing a common container in Grafana. 
+
+:::caution
+
+Query responses are often more than one single data frame. 
+
+:::
+
+The data frame a column-oriented table (the _fields_) with metadata (the _frame_) attached. Since data frame columns can have labels attached to them (`key=value`, `key2=val`...), it can hold Prometheus like responses as well. 
+
+Each field in a data frame contains optional information about the values in the field, such as units, scaling, and so on. By adding field configurations to a data frame, Grafana can configure visualizations automatically. For example, you could configure Grafana to automatically set the unit provided by the data source.
+
+## Data planes overview  
+
+The data plane adds a property layer to the frame as metadata. It indicates the data frame _type_ (for example, a timeseries or a heatmap), which consists of a _kind_ (of data) and its _format_ (Prometheus-like, SQL-table-like).
+
+:::tip
+
+Data plane types are to data frames what TypeScript is to JavaScript.
+
+:::
+
+The data plane contract is a written set of rules that explain how producers of data (datasources, transformations) must form the frames, and how data consumers (like dashboards, alerting, and apps) can expect the data they receive to be like. In short, it describes the rules for valid and invalid schemas for each data frame type.
+
+## Why use data planes?
+
+Although the use of data planes is not enforced, the main objective of the data plane is to make Grafana more self-interoperable between data sources and features like dashboards and alerting. 
```

Comment:
This is generally true, although not true of labeled data when using SQL expressions. In that feature it is enforced. Only the type though, not the version. maybe "is usually not enforced (a current exception being SQL expressions)"? is enough detail for here?